operations/rack_resiliency/rack_resiliency.md

```diff
@@ -0,0 +1,490 @@
+
+* Determining whether the failure is isolated to a single node or part of a rack-level failure.
+* Monitors the health and status of critical services, and logs alerts if any imbalance is detected following a failure.
+* Verifying placement and availability of management nodes, including both `Kubernetes` control plane and `Ceph` storage nodes.
+
+**Note:** Any modifications to critical services may take up to 10 minutes to reflect in CLI command outputs and to appear in the monitored list.
+
+### 5 RR CLI
+
+The CLI for interfacing with rack resiliency service is part of `craycli`. A new module (`rrs`) is added to `craycli` to support rack resiliency specific commands. The backend for the CLI which serves REST endpoints is provided as a container along with the rack resiliency monitor in the same pod.
+
+#### Commands 
+
+The following commands are supported as two groups:
+
+**zones:** to list / describe `Kubernetes` topology zones and `Ceph` zones as configured
+
+**critical services:** to list/describe critical services defined in the RMS ConfigMap
+
+The sub commands for each of the groups are used to list, describe and update resources associated with the groups.
+
+RRS `craycli` commands
+
+```bash
+    cray rrs zones list
+    cray rrs zones describe 
+    cray rrs criticalservices list
+    cray rrs criticalservices describe
+    cray rrs criticalservices update --from-file <file-path>
+```
+
+Please refer https://github.com/Cray-HPE/docs-csm/blob/CASM-5612/operations/rack_resiliency/rack_resiliency_cli.md for more information on viewing/managing zones (`Kubernetes` and `Ceph`) and critical service via RR CLI.
+
+## Steps to achieve RR
+
+### Install/ Upgrade
+
+#### 1 To enable Rack Resiliency feature
+
+By default the Rack Resiliency feature will be disabled and the user/admin can enable the Rack Resiliency feature by following the below steps.
+
+**Warning:** The Rack Resiliency feature flag has to be enabled before the CSM installation or upgrade is performed.
+Enabling this feature will create zones for the master, worker and storage management nodes.
+
+**Procedure**
+
+Update the `customizations.yaml` file before commencing the CSM installation or upgrade.
+
+Update the `spec.services.rack-resiliency.enabled` flag from `false` to `true` in the `customizations.yaml` file. This `customizations.yaml` file will be present in the root directory of the CSM tarball.
+
+During the CSM installation or upgrade, a `Kubernetes` secret named `site-init` under `loftsman` will be created with the `spec.services.rack-resiliency.enabled` flag enabled. Further, the same secret will be checked for zone creation for Master, Worker and Storage nodes. If the flag is in enabled state, zones will be created else the zones will not be created for Master, Worker and Storage nodes.
+
+#### 2 To define the `Kubernetes` and `Ceph` zone prefixes

```

Comment:
This section, step 2 in the procedure, says it has to happen before the install or upgrade is performed. But the final parts of step 1 talk about things being done during the install or upgrade. Is this really the next step after the above section, in that case?